### PR TITLE
Initial game mechanics

### PR DESCRIPTION
**Pull request by LoganTHarvell**
Work in progress [WIP]. Do not merge until completed.

Branch focuses on the initial game mechanics, specifically basic resource mechanics surrounding fuel, mining, farming, and population.

Completed:
- Added fuel based movement
- Added minerals -> fuel transfer during docking
- Added basic mining mechanics
- Added basic farming mechanics
- Add population growth mechanics with infrastructure development

Bug Fixes and Adjustments:
- Modified UI color scheme
- Modified gameScreen graphics/font
- Fixed planet identical coordinates bug
- Fixed PlanetInfo display conditions
- Added elapsedTime and frame count to gameState
- Added time textbox to gameScreen UI

### CODE REVIEW

**Comment by LoganTHarvell** [comment]
Yes I am crazy, this branch might as well be done. All the initial game mechanics are working decent enough for the game to be playable, albeit pretty slow and somewhat shitty. 

There are some issues and tweaks we will need, but I'll make a separate issues post. Feel free to review and merge.

File: Astral Pathfinder/Game/PlanetManager.cpp [hunk 1]
```diff
@@ -103,22 +106,27 @@ Planet PlanetManager::getSelectedPlanet() {
...
   return planets[dockedPlanetIndex];
 }
 
 void PlanetManager::setPlanetDepoPercent(int p, int flag) {
   if(flag == 1)
-    planets[dockedPlanetIndex].setDepositsPercent(p);
+    planets[dockedPlanetIndex].setMiningPercent(p);
   if(flag == 2)
-    planets[selectedPlanetIndex].setDepositsPercent(p);
+    planets[selectedPlanetIndex].setMiningPercent(p);
```
Comment (holdemani) [inline Astral Pathfinder/Game/PlanetManager.cpp:122]: If you're going to change these, change the current function name as well

File: Astral Pathfinder/Game/UIManager.cpp [hunk 2]
```diff
@@ -9,16 +9,20 @@
 // MARK: Header File
 #include "UIManager.hpp"
 
+#include <string>
```
Comment (holdemani) [inline Astral Pathfinder/Game/UIManager.cpp]: // MARK: Libraries and Frameworks

You're slacking kid

File: Astral Pathfinder/UI/PlanetInfo.cpp [hunk 3]
```diff
@@ -79,15 +79,19 @@ void PlanetInfo::clean() {
...
 
 void PlanetInfo::setUiTextures(Planet p) {
+  using namespace SliderParameters;
+  
   setBoxes(p);
   
-  if (!sliderOne.isInitialized() && !sliderTwo.isInitialized() && p.getPopulation() != 0) {
-    sliderOne.setTextures(p.getFertilityPercent());
+  if (!sliderOne.isInitialized() && !sliderTwo.isInitialized()
+      && (p.getPopulation() > 0 || p.playerIsDocked())) {
```
Comment (holdemani) [inline Astral Pathfinder/UI/PlanetInfo.cpp]: Was this necessary?

**Review by holdemani — DISMISSED** [review]
See comments before I merge

**Comment by holdemani** [comment]
Additionally, homeworld starts with 50/50 mining/farming and 0/100 infra/reserve, but a newly colonized planet has 100/0 mining/farming and 50/50 infra and reserve. is this intentional?

File: Astral Pathfinder/Game/PlanetManager.cpp (hunk 1 above)
Comment (LoganTHarvell) [reply Astral Pathfinder/Game/PlanetManager.cpp:122]: Done. Feel free to rename things if you see more.

File: Astral Pathfinder/UI/PlanetInfo.cpp (hunk 3 above)
Comment (LoganTHarvell) [reply Astral Pathfinder/UI/PlanetInfo.cpp]: Fixed.

**Comment by LoganTHarvell** [comment]
Revised and ready to merge.

**Pull request merged**